### PR TITLE
use importas linter to check import aliases.

### PR DESCRIPTION
**Pull request by bitoku**
### Which issue this PR addresses:

Fixes [ARO 1693](https://issues.redhat.com/browse/ARO-1693)

### What this PR does / why we need it:

This replaces the current import alias check to (importas)[https://golangci-lint.run/usage/linters/#importas].

There are some caveats.

1. with `no-unaliased` it forbids *all* packages to be unaliased, but can't forbid/allow specific packages to be unaliased.
2. It can't check underscore import (see https://github.com/julz/importas/blob/e9218386d3f4cd0421d099009f2fdc209269fc22/analyzer.go#L60)

### Test plan for issue:

This is a part of CI and no need to test.
I checked the behaviour in my local machine.

### Is there any documentation that needs to be updated for this PR?

No, tech debt cleanup.

### CODE REVIEW

**Review by facchettos — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -13,6 +13,18 @@ issues:
...
   # which shadows all the ST* checks. Refer https://github.com/golangci/golangci-lint/issues/2281
+  - path: "pkg/client/(.+)\\.go"
+    linters:
+      - importas
+  - path: "pkg/database/cosmosdb/zz_generated_(.+)\\.go"
+    linters:
+      - importas
+  - path: "pkg/operator/(apis|clientset|mocks)/(.+)\\.go"
+    linters:
+      - importas
```
Comment (SudoBrendan) [inline .golangci.yml:24]: I understand `clientset` and `mocks`, but not `apis`. What's the reason we don't want those linted?

**Review by SudoBrendan — DISMISSED** [review]
Ive got a few concerns about the current exclusions, but otherwise LGTM.

I have to admit, when I first saw the large `alias` block, I was originally thinking "this looks like toil if I have to add an `alias` line for nearly every package we use", then I hit our custom linter code and I was like "well, it's way better than that!". What processes do we need to adopt to ensure that new libraries land in the linter? I feel at the very least this PR deserves a communication to the team at-large.

Good work here, thanks!

File: .golangci.yml (hunk 1 above)
Comment (bitoku) [reply .golangci.yml:24]: I just reflected here https://github.com/Azure/ARO-RP/blob/be4bae3d0ce01c2c9ba3cc7fe7bf30b0a1cd78c2/hack/validate-imports/validate-imports.go#L155 .
but you are true.
After removing it, it didn't output error.
I'll fix this.

**Comment by facchettos** [comment]
@SudoBrendan I disagree with you here, there isn't currently a process to make sure libraries are added to the current custom linter, and it's also an optional thing as not all libraries have rules. We can discuss about a process but it shouldn't be related to that specific PR

**Comment by bitoku** [comment]
I have the same opinion as @facchettos .
There is [`no-extra-aliases`](https://golangci-lint.run/usage/linters/#importas) option to make sure all libraries to be checked.
but there are already many packages that are not included in `validate-import.go` and they will cause an error if we set the option.

<details><summary>golangci-lint with no-extra-aliases</summary>
<p>

```
❯ golangci-lint run
pkg/util/encryption/aescbc_test.go:13:2: import "github.com/Azure/ARO-RP/test/util/error" has alias "utilerror" which is not part of config (importas)
        utilerror "github.com/Azure/ARO-RP/test/util/error"
        ^
pkg/util/encryption/multi_test.go:14:2: import "github.com/Azure/ARO-RP/test/util/error" has alias "utilerror" which is not part of config (importas)
        utilerror "github.com/Azure/ARO-RP/test/util/error"
        ^
pkg/util/encryption/xchacha20poly1305_test.go:13:2: import "github.com/Azure/ARO-RP/test/util/error" has alias "utilerror" which is not part of config (importas)
        utilerror "github.com/Azure/ARO-RP/test/util/error"
        ^
pkg/operator/controllers/alertwebhook/alertwebhook_controller_test.go:16:2: import "sigs.k8s.io/controller-runtime/pkg/client/fake" has alias "ctrlfake" which is not part of config (importas)
        ctrlfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
        ^
pkg/util/azureclient/applens/applens_client_test.go:19:2: import "github.com/Azure/ARO-RP/test/util/http" has alias "testhttp" which is not part of config (importas)
        testhttp "github.com/Azure/ARO-RP/test/util/http"
        ^
pkg/util/azureclient/applens/applens_error_test.go:18:2: import "github.com/Azure/ARO-RP/test/util/http" has alias "testhttp" which is not part of config (importas)
        testhttp "github.com/Azure/ARO-RP/test/util/http"
        ^
pkg/operator/controllers/ingress/ingress_controller_test.go:16:2: import "sigs.k8s.io/controller-runtime/pkg/client/fake" has alias "ctrlfake" which is not part of config (importas)
        ctrlfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
        ^
pkg/util/scheme/scheme.go:22:2: import "k8s.io/kubernetes/pkg/apis/apps/v1" has alias "appsv1defaults" which is not part of config (importas)
        appsv1defaults "k8s.io/kubernetes/pkg/apis/apps/v1"
        ^
pkg/util/scheme/scheme.go:23:2: import "k8s.io/kubernetes/pkg/apis/core/v1" has alias "corev1defaults" which is not part of config (importas)
        corev1defaults "k8s.io/kubernetes/pkg/apis/core/v1"
        ^
pkg/util/scheme/scheme.go:24:2: import "k8s.io/kubernetes/pkg/apis/rbac/v1" has alias "rbacv1defaults" which is not part of config (importas)
        rbacv1defaults "k8s.io/kubernetes/pkg/apis/rbac/v1"
        ^
pkg/util/azureclient/authz/remotepdp/client_test.go:16:2: import "github.com/Azure/ARO-RP/test/util/http" has alias "testhttp" which is not part of config (importas)
        testhttp "github.com/Azure/ARO-RP/test/util/http"
        ^
test/e2e/admin_portal.go:13:2: import "github.com/serge1peshcoff/selenium-go-conditions" has alias "conditions" which is not part of config (importas)
        conditions "github.com/serge1peshcoff/selenium-go-conditions"
        ^
test/e2e/operator.go:22:2: import "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers" has alias "cov1Helpers" which is not part of config (importas)
        cov1Helpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
        ^
test/e2e/operator.go:30:2: import "github.com/Azure/ARO-RP/pkg/operator/controllers/imageconfig" has alias "imageController" which is not part of config (importas)
        imageController "github.com/Azure/ARO-RP/pkg/operator/controllers/imageconfig"
        ^
test/e2e/setup.go:29:2: import "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned" has alias "monitoringclient" which is not part of config (importas)
        monitoringclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
        ^
pkg/api/validate/dynamic/serviceprincipal_test.go:13:2: import "github.com/golang-jwt/jwt/v4" has alias "jwt" which is not part of config (importas)
        jwt "github.com/golang-jwt/jwt/v4"
        ^
pkg/frontend/frontend.go:18:2: import "github.com/go-chi/chi/v5/middleware" has alias "chiMiddlewares" which is not part of config (importas)
        chiMiddlewares "github.com/go-chi/chi/v5/middleware"
        ^
pkg/hive/manager_test.go:17:2: import "k8s.io/client-go/kubernetes/fake" has alias "kubernetesfake" which is not part of config (importas)
        kubernetesfake "k8s.io/client-go/kubernetes/fake"
        ^
pkg/monitor/cluster/debugpods.go:17:2: import "k8s.io/apimachinery/pkg/util/validation" has alias "kuval" which is not part of config (importas)
        kuval "k8s.io/apimachinery/pkg/util/validation"
        ^
pkg/operator/controllers/checkers/clusterdnschecker/checker_test.go:12:2: import "sigs.k8s.io/controller-runtime/pkg/client/fake" has alias "ctrlfake" which is not part of config (importas)
        ctrlfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
        ^
pkg/operator/controllers/checkers/serviceprincipalchecker/checker_test.go:13:2: import "github.com/golang-jwt/jwt/v4" has alias "jwt" which is not part of config (importas)
        jwt "github.com/golang-jwt/jwt/v4"
        ^
pkg/operator/controllers/node/node_controller_test.go:24:2: import "github.com/Azure/ARO-RP/test/util/conditions" has alias "utilconditions" which is not part of config (importas)
        utilconditions "github.com/Azure/ARO-RP/test/util/conditions"
        ^
pkg/portal/cluster/statistics.go:12:2: import "github.com/prometheus/client_golang/api" has alias "prometheusAPI" which is not part of config (importas)
        prometheusAPI "github.com/prometheus/client_golang/api"
        ^
pkg/portal/cluster/statistics.go:13:2: import "github.com/prometheus/client_golang/api/prometheus/v1" has alias "v1" which is not part of config (importas)
        v1 "github.com/prometheus/client_golang/api/prometheus/v1"
        ^
pkg/util/clusterauthorizer/authorizer_test.go:13:2: import "github.com/golang-jwt/jwt/v4" has alias "jwt" which is not part of config (importas)
        jwt "github.com/golang-jwt/jwt/v4"
        ^
pkg/util/conditions/controllerconditions_test.go:20:2: import "github.com/Azure/ARO-RP/test/util/conditions" has alias "utilconditions" which is not part of config (importas)
        utilconditions "github.com/Azure/ARO-RP/test/util/conditions"
        ^
```

</p>
</details> 

It's true that how to ensure that should be discussed, but I think it can be discussed aside from this PR since this PR aims for replacing the part of complex `validate-import.go`.

**Comment by bitoku** [comment]
@SudoBrendan What do you think?

**Review by facchettos — APPROVED** [review]

**Pull request merged**